File: engine/model/src/main/scala/tools/packet/pet/PetFoodResponse.scala
package tools.packet.pet

import net.opcodes.SendOpcode
import tools.packet.PacketInput

class PetFoodResponse(private var _characterId: Int, private var _index: Byte, private var _success: Boolean, private var _balloonType: Boolean) extends PacketInput {
  def characterId: Int = _characterId

  def index: Byte = _index

  def success: Boolean = _success

  def balloonType: Boolean = _balloonType

  override def opcode(): SendOpcode = SendOpcode.PET_COMMAND
}

File: engine/engine-model/src/main/java/tools/packet/pet/PetFoodResponse.java
package tools.packet.pet;

import net.opcodes.SendOpcode;
import tools.packet.PacketInput;

public record PetFoodResponse(Integer characterId, Byte index, Boolean success,
                              Boolean balloonType) implements PacketInput {
   @Override
   public SendOpcode opcode() {
      return SendOpcode.PET_COMMAND;
   }
}